ui/src/app/utils/urls.ts

```diff
@@ -9,3 +9,23 @@ export function canonicalizeUrl(url: string): string {
   // and https:// may yield a bad certificate if not configured.
   return `http://${url}`;
 }
+
+function isValidUrl(potentialUrl: string): boolean {
+  try {
+    new URL(potentialUrl);
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+export function getCustomOrDefaultUrl(customUrl: string, defaultUrl: string) {
```

Comment:
I am unsure of the name of this function. I am open to suggestions if anyone can think of anything better.